api/src/main/java/uk/gov/hmcts/fees2/register/api/controllers/FeeVersionController.java

```diff
@@ -93,8 +95,8 @@ public void approve(@PathVariable("feeCode") String feeCode, @PathVariable("vers
     })
     @PatchMapping("/fees/{feeCode}/versions/{version}/reject")
     @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void reject(@PathVariable("feeCode") String feeCode, @PathVariable("version") Integer version) {
-        feeVersionService.changeStatus(feeCode, version, FeeVersionStatus.draft, null);
+    public void reject(@PathVariable("feeCode") String feeCode, @PathVariable("version") Integer version, @RequestBody(required = false) ReasonDto reasonDto) {
+            feeVersionService.changeStatus(feeCode, version, FeeVersionStatus.draft, null, (null != reasonDto && null!= reasonDto.getReasonForReject()) ? Encode.forHtml(reasonDto.getReasonForReject()) : "");
```

Comment:
if rejection reason is null in request, we should keep this null or unchanged for the 1st request but if there was a previous reason then following reasons in request should override the previous.

Also take care of code formatting

Reply:
Updated logic with null input and formatted code